fastlane_bot/modes/triangle_bancor_v3_two_hop.py

```diff
@@ -8,362 +8,54 @@
-            external_curves = list(set(external_curves))
-            carbon_curves = [
-                curve
-                for curve in external_curves
-                if curve.params.get("exchange") in self.ConfigObj.CARBON_V1_FORKS
-            ]
-            external_curves = [
-                curve
-                for curve in external_curves
-                if curve.params.get("exchange") not in self.ConfigObj.CARBON_V1_FORKS
-            ]
-            if not external_curves and not carbon_curves:
+            bancor_v3_curve_0 = self.CCm.bypairs(f"{self.ConfigObj.BNT_ADDRESS}/{tkn0}").byparams(exchange="bancor_v3").curves
+            if len(bancor_v3_curve_0) == 0:
                 continue
 
-            bancor_v3_curve_0 = (
-                self.CCm.bypairs(f"{T.BNT}/{tkn0}")
-                .byparams(exchange="bancor_v3")
-                .curves
-            )
-            bancor_v3_curve_1 = (
-                self.CCm.bypairs(f"{T.BNT}/{tkn1}")
-                .byparams(exchange="bancor_v3")
-                .curves
-            )
-            if bancor_v3_curve_0 is None or bancor_v3_curve_1 is None:
-                continue
-            if len(bancor_v3_curve_0) == 0 or len(bancor_v3_curve_1) == 0:
+            bancor_v3_curve_1 = self.CCm.bypairs(f"{self.ConfigObj.BNT_ADDRESS}/{tkn1}").byparams(exchange="bancor_v3").curves
+            if len(bancor_v3_curve_1) == 0:
                 continue
 
-            miniverses = []
-            if len(external_curves) > 0:
-                for curve in external_curves:
-                    miniverses += [bancor_v3_curve_0 + bancor_v3_curve_1 + [curve]]
-            if len(carbon_curves) > 0:
+            base_dir_one = [curve for curve in carbon_curves if curve.pair == carbon_curves[0].pair]
+            base_dir_two = [curve for curve in carbon_curves if curve.pair != carbon_curves[0].pair]
+            miniverses = [bancor_v3_curve_0 + bancor_v3_curve_1 + [curve] for curve in external_curves]
 
-                if len(carbon_curves) > 0:
-                    base_direction_pair = carbon_curves[0].pair
-                    base_direction_one = [curve for curve in carbon_curves if curve.pair == base_direction_pair]
-                    base_direction_two = [curve for curve in carbon_curves if curve.pair != base_direction_pair]
+            if len(base_dir_one) > 0:
+                miniverses += [bancor_v3_curve_0 + bancor_v3_curve_1 + base_dir_one]
 
-                    if len(base_direction_one) > 0:
-                        miniverses += [bancor_v3_curve_0 + bancor_v3_curve_1 + base_direction_one]
+            if len(base_dir_two) > 0:
+                miniverses += [bancor_v3_curve_0 + bancor_v3_curve_1 + base_dir_two]
 
-                    if len(base_direction_two) > 0:
-                        miniverses += [bancor_v3_curve_0 + bancor_v3_curve_1 + base_direction_two]
+            miniverses += [bancor_v3_curve_0 + bancor_v3_curve_1 + carbon_curves]
```

Comment:
This should be removed as the carbon curves are being added separately above.

Reply:
Fixed